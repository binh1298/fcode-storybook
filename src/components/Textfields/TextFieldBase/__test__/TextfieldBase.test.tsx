import TextfieldBase, { TexfieldBaseProps } from "../";

import { render, RenderResult, screen } from "@testing-library/react";

describe("<TextfieldBase />", () => {
    let wrapper: RenderResult;

    const propsForSecondaryColor: TexfieldBaseProps = {
        label: "Test",
        value: "Value",
        color: "secondary",
    };

    const propsForPrimaryColor: TexfieldBaseProps = {
        label: "Test",
        value: "Value",
        color: "primary",
    };

    const propsWithErrorMessage: TexfieldBaseProps = {
        label: "Test",
        value: "Value",
        errorMessage: "error",
        color: "secondary",
    };

    it("should match snapshot", () => {
        wrapper = render(<TextfieldBase {...propsForSecondaryColor} />);
        expect(wrapper.container).toMatchSnapshot();
    });

    it("should match snapshot", () => {
        wrapper = render(<TextfieldBase {...propsForPrimaryColor} />);
        expect(wrapper.container).toMatchSnapshot();
    });

    it("should match snapshot", () => {
        wrapper = render(<TextfieldBase {...propsWithErrorMessage} />);
        expect(wrapper.container).toMatchSnapshot();
    });
});
