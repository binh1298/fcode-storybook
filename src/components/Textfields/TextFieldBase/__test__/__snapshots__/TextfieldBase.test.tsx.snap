// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextfieldBase /> should not show error color when not have any error message 1`] = `
<div>
  <div
    class="MuiFormControl-root MuiTextField-root makeStyles-root-17 makeStyles-root-21 css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
  >
    <label
      class="makeStyles-cssLabel-15 makeStyles-cssLabel-19 MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary MuiFormLabel-filled css-1sagrvw-MuiFormLabel-root-MuiInputLabel-root"
      data-shrink="true"
    >
      Test
    </label>
    <div
      class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl makeStyles-fontColor-16 makeStyles-fontColor-20 css-1u1j92g-MuiInputBase-root-MuiOutlinedInput-root"
    >
      <input
        aria-invalid="false"
        class="MuiOutlinedInput-input MuiInputBase-input css-1trjin0-MuiInputBase-input-MuiOutlinedInput-input"
        type="text"
        value="Value"
      />
      <fieldset
        aria-hidden="true"
        class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
      >
        <legend
          class="css-186xcr5"
        >
          <span>
            Test
          </span>
        </legend>
      </fieldset>
    </div>
  </div>
</div>
`;

exports[`<TextfieldBase /> should show error color when errorMessage is available 1`] = `
<div>
  <div
    class="MuiFormControl-root MuiTextField-root makeStyles-root-10 makeStyles-root-14 css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
  >
    <label
      class="makeStyles-cssLabel-8 makeStyles-cssLabel-12 MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary Mui-error MuiFormLabel-filled css-1sagrvw-MuiFormLabel-root-MuiInputLabel-root"
      data-shrink="true"
    >
      Test
    </label>
    <div
      class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary Mui-error MuiInputBase-formControl makeStyles-fontColor-9 makeStyles-fontColor-13 css-1u1j92g-MuiInputBase-root-MuiOutlinedInput-root"
    >
      <input
        aria-invalid="true"
        class="MuiOutlinedInput-input MuiInputBase-input css-1trjin0-MuiInputBase-input-MuiOutlinedInput-input"
        type="text"
        value="Value"
      />
      <fieldset
        aria-hidden="true"
        class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
      >
        <legend
          class="css-186xcr5"
        >
          <span>
            Test
          </span>
        </legend>
      </fieldset>
    </div>
    <p
      class="MuiFormHelperText-root Mui-error MuiFormHelperText-sizeMedium MuiFormHelperText-contained MuiFormHelperText-filled css-1id7my8-MuiFormHelperText-root"
    >
      error message
    </p>
  </div>
</div>
`;
